backend/backend-application/src/main/java/eu/socialedge/hermes/backend/application/api/service/PublicationService.java

```diff
@@ -0,0 +1,136 @@
+import eu.socialedge.hermes.backend.publication.domain.Publication;
+import eu.socialedge.hermes.backend.publication.repository.PublicationRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import eu.socialedge.hermes.backend.schedule.repository.ScheduleRepository;
+import eu.socialedge.hermes.backend.timetable.domain.Folder;
+import eu.socialedge.hermes.backend.timetable.domain.TimetableGenerationService;
+import eu.socialedge.hermes.backend.transit.domain.infra.StationRepository;
+import eu.socialedge.hermes.backend.transit.domain.service.LineRepository;
+import lombok.val;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class PublicationService extends PagingAndSortingService<Publication, String, PublicationDTO> implements PublicationsApiDelegate {
+
+    private static final String PDF_EXTENSION = "pdf";
+    private static final String ZIP_EXTENSION = "zip";
+    private static final String MEDIATYPE_HEADER = "application/%s";
+    private static final String FILE_NAME_TEMPLATE = "%s.%s";
+
+    private final TimetableGenerationService timetableGenerationService;
+    private final LineRepository lineRepository;
+    private final ScheduleRepository scheduleRepository;
+    private final StationRepository stationRepository;
+
+    @Autowired
+    public PublicationService(PublicationRepository repository, PublicationMapper mapper, TimetableGenerationService timetableGenerationService,
+                              LineRepository lineRepository, ScheduleRepository scheduleRepository, StationRepository stationRepository) {
+        super(repository, mapper);
+        this.timetableGenerationService = timetableGenerationService;
+        this.lineRepository = lineRepository;
+        this.scheduleRepository = scheduleRepository;
+        this.stationRepository = stationRepository;
+    }
+
+    @Override
+    public ResponseEntity<List<PublicationDTO>> listAllPublications(Integer size, Integer page, String sort, String filtering) {
+        return list(size, page, sort, filtering);
+    }
+
+    @Override
+    public ResponseEntity<PublicationDTO> getPublication(String id) {
+        return get(id);
+    }
+
+    @Override
+    public ResponseEntity<Resource> getPublicationContent(String id) {
+        val publication = repository.findOne(id);
+        if (publication == null) {
+            return ResponseEntity.notFound().build();
+        }
+        val fileType = publication.getStation() != null && publication.getLine() != null ? PDF_EXTENSION : ZIP_EXTENSION;
```

Comment:
mmm...I would say it's neither reliable nor scalable way to detect file content type. Metadata must be as close as possible to the thing it describes i.e. type should be in publication. It was proposed to create a File abstraction instead of Document, that will contain mime type.

Instead of `Publication#byte[] file`:
```
class File {
   String type;
   byte[] data;
}
class Publication {
   //...
   File file;
   //...
}
```
You may also create enum for File:
```
enum MediaType {
   ZIP("application/zip"), PDF("application/pdf");
   String mime;
   // constructor
   String mime() {return mime;}
}
```
so that File will look like:
```
class File {
   MediaType type;
   byte[] data;
}
```
and just `headers.setContentType(file.type().mime());`

Reply:
MediaType with file extension:
```
enum MediaType {
   ZIP("application/zip", "zip"), PDF("application/pdf", "pdf");
   String mime;
   String extension;
   // constructor
   String mime() {return mime;}
   String extension() {return extension;}
}
```